pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -84,13 +86,64 @@ type Client struct {
 
 	// The gateway-api client we will use to access objects
 	gatewayAPIClient gatewayapiclient.Interface
+
+	// initialSync is set to true after performing an initial processing of all objects.
+	initialSync *atomic.Bool
 }
 
 var _ model.ConfigStoreCache = &Client{}
 
+func New(client kube.Client, revision, domainSuffix string) (model.ConfigStoreCache, error) {
+	schemas := collections.Pilot
+	if features.EnableServiceApis {
+		schemas = collections.PilotServiceApi
+	}
+	return NewForSchemas(context.Background(), client, revision, domainSuffix, schemas)
+}
+
+func NewForSchemas(ctx context.Context, client kube.Client, revision, domainSuffix string, schemas collection.Schemas) (model.ConfigStoreCache, error) {
+	out := &Client{
+		domainSuffix:     domainSuffix,
+		schemas:          schemas,
+		revision:         revision,
+		queue:            queue.NewQueue(1 * time.Second),
+		kinds:            map[config.GroupVersionKind]*cacheHandler{},
+		istioClient:      client.Istio(),
+		gatewayAPIClient: client.GatewayAPI(),
+		initialSync:      atomic.NewBool(false),
+	}
+
+	known, err := knownCRDs(ctx, client.Ext())
+	if err != nil {
+		return nil, err
+	}
+	for _, s := range schemas.All() {
+		// From the spec: "Its name MUST be in the format <.spec.name>.<.spec.group>."
+		name := fmt.Sprintf("%s.%s", s.Resource().Plural(), s.Resource().Group())
+		if _, f := known[name]; f {
+			var i informers.GenericInformer
+			var err error
+			if s.Resource().Group() == "networking.x-k8s.io" {
+				i, err = client.GatewayAPIInformer().ForResource(s.Resource().GroupVersionResource())
+			} else {
+				i, err = client.IstioInformer().ForResource(s.Resource().GroupVersionResource())
+			}
+			if err != nil {
+				return nil, err
+			}
+			out.kinds[s.Resource().GroupVersionKind()] = createCacheHandler(out, s, i)
+		} else {
+			scope.Warnf("Skipping CRD %v as it is not present", s.Resource().GroupVersionKind())
+			out.schemas = out.schemas.Remove(s)
+		}
+	}
+
+	return out, nil
+}
+
 // Validate we are ready to handle events. Until the informers are synced, we will block the queue
 func (cl *Client) checkReadyForEvents(curr interface{}) error {
```

Comment:
Should the `*Client`'s `HasSynced` method check this? I think we use that when deciding if the `istiod` is ready.

And if it does, I _don't_ think the "workloadentry only" ones we create for remote clusters block readiness – they probably should – and if they do we need to handle the remote cluster sync timeout (https://github.com/istio/istio/pull/31771).

Reply:
@stevenctl Do you mean replace `cl.informerSynced()` with `cl.HasSynced ()` in L146? 
no, then the new events during SyncAll are ignored